src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs

```diff
@@ -91,6 +82,24 @@ private static int RunMcg(NativeCompileSettings config)
             mcgArgs.Add("--outputpath");
             mcgArgs.Add(outPath);
 
+            var ilSdkPath = Path.Combine(config.IlcSdkPath, "sdk");
+            foreach (string refPath in Directory.EnumerateFiles(ilSdkPath, "*.dll"))
+            {
+
+                mcgArgs.Add("--r");
+                mcgArgs.Add(refPath);
+            }
+
+            foreach (string refPath in Directory.EnumerateFiles(config.AppDepSDKPath, "*.dll"))
+            {
+                // System.Runtime.Extensions define an internal type called System.Runtime.InteropServices.Marshal which 
+                // conflicts with Marshal from S.P.Interop , we don't need System.Runtime.Extensions anyways,skip it.
+                if (refPath.Contains("System.Runtime.Extensions.dll")) continue;
```

Comment:
AppDep SDK is a point in time thing and is expected to go-away. Mcg do not understand packaged now , so it cant consume ILSDK package directly.But it's something which will be fixed once mcg is out in public.